src/HomogeneousCalorimeter_geo.cpp

```diff
@@ -406,58 +406,114 @@ static std::tuple<int, int> add_disk(Detector& desc, Assembly& env, xml::Collect
 static std::tuple<int, int> add_12surface_disk(Detector& desc, Assembly& env, xml::Collection_t& plm,
                                                SensitiveDetector& sens, int sid)
 {
```

Comment:
I was thinking how to arrange the module tightly along the volume border, but I realize it would be difficult math problem and not possible/very difficult to fill the rectangle in the polygon. So, I used the defaulted fillrectangle function and fill the gap manually. I know it is not convenient for people want to parameterized the size of module for some tests as they will just get overlap issue with bigger module. I think, instead of parameterizing the position of modules at edge, I could built a boolean flag for people to turn modules at the edge on or off.

Reply:
This is still unresolved. You took a general geometry plugin allowing for general specification homogeneous calorimeters, and you turned it into the N EEMC plugin (for good reason, I don't have any quibble with that). There are potentially other subsystems using this plugin, or considering it (there are far foward calorimeters that don't need more than the general approach). It's not maintainable to keep adding NEEMC details to the general cpp plugin. Please split off your changes to a different source file with only the parts you are actually using, and name it so it is clear that it is specific to the negative endcap.

Reply:
I got it. Should I keep this specified plugin in the general NEEMC file when I split off this configuration?

Reply:
@wdconinc Could I make another pull request for this? This pull request is about the updated and improved geometry close to beam pipe. And the work of spin off the this configuration from the homogenouscalorimeter would be a lots and has nothing to do with this pull request. Could I merge this into the main if there is no problem other than making another geometry plugin for NEEMC?

Reply:
And sorry that I was not able to think about this recently, but I will open the new pull request about this, if you agreed with me, and back to finish this.

Reply:
Thanks for giving another solution for this conversation. I change the way to assign cell ID in my last commit by using the row and column id instead of the sequential ID. @wdconinc could you help me do another review. Thanks!